### PR TITLE
Changed configuration paths and behaviours

### PR DESCRIPTION
**Pull request by Ev1lbl0w**
This PR changes the way Komorebi stores it's configuration files, and where it retrieves wallpapers.

**TL;DR**: Komorebi now loads config and custom properties from `~/.config/komorebi` *(the most likely path)*, and wallpaper creator no longer needs root permission to save wallpapers.

I made three big changes, each with it's own commit:

- **Changed default configuration path**: Changed the path from the hardcoded `~/.Komorebi.prop` to a more appropriate place. Komorebi now stores everything in a new path, which is, by order of preference:
  - `$XDG_CONFIG_HOME/komorebi`
  - `$HOME/.config/komorebi`
  - `$HOME/komorebi`
It also detects the old configuration file and ports it to the new file structure, so this change is "invisible" to the end user. The new place is now `<config dir>/komorebi/komorebi.prop`; I chose to preserve the old name, but feel free to suggest another name for the file.
- **Added new path for custom wallpapers inside config folder**: Komorebi now loads wallpapers from the new config folder as well; this allows to easily separate "official" wallpapers from the custom ones the user made. Plus the user has access to those files without requiring root access.
- **Changed wallpaper creator to save at new config path**: Building upon the last commit, the wallpaper creator app now stores the wallpaper at the config folder, and no longer requires root permissions for it; thus, Komorebi now loads these custom wallpapers immediately, and it's **way** easier and less dangerous for the end user.

**Please do not merge this PR before the changes to the build system/FHS paths _(aka #14)_ are merged; I intend to rebase this branch on top of it. I'm opening this PR only for reviewing and potential discussion.**

### CODE REVIEW

**Comment by Ev1lbl0w** [comment]
Travis fails because Ubuntu 16.04's version of libglib2 is [too old](https://packages.ubuntu.com/xenial/libglib2.0-0). Should we upgrade to something like Ubuntu 18.04 `bionic`, or is there a way to upgrade the lib?

**Comment by Kangie** [comment]
> Travis fails because Ubuntu 16.04's version of libglib2 is [too old](https://packages.ubuntu.com/xenial/libglib2.0-0). Should we upgrade to something like Ubuntu 18.04 `bionic`, or is there a way to upgrade the lib?

It'll be fixed with the meson build changes - I've bumped travis to Bionic there.

**Comment by Kangie** [comment]
I like it. You've already covered my immediate thought (whether or not we should automatically save new wallpapers to the local dir) - the current 'paste this in your terminal' is a rough edge that we can do without.

Do you think that it's worth looking at giving the option to save to the system-wide wallpaper directory when the wallpaper is generated? At some point we're going to have to address modifying wallpapers saved there if we want to allow users to edit the premade wallpapers.

Nothing else to add here - your work so far looks pretty good and well thought out. Hopefully the Meson / FHS changes don't cause you too much additional work.

**Comment by Ev1lbl0w** [comment]
> Do you think that it's worth looking at giving the option to save to the system-wide wallpaper directory when the wallpaper is generated? At some point we're going to have to address modifying wallpapers saved there if we want to allow users to edit the premade wallpapers.

My immediate thought would be to show a "popup" when editing an official wallpaper, prompting the user that to be able to modify the wallpaper, a copy of it would be made to the custom wallpaper directory; this would also have the added benefit of preserving the original wallpaper.

**Comment by Kangie** [comment]
So the logic would be: 

1. Select wallpaper for editing
2. If built-in, copy to local wallpaper dir
3. Edit wallpaper config

Assuming there's some case handling to prefer a wallpaper in the local dir where two wallpapers of the same name are found (so masking the shipped copy essentially).

**Comment by Ev1lbl0w** [comment]
Yes, that's what I had in mind. We could, in the UI, show a separation between "Custom" and "Official" wallpapers, to be less confusing to the user, and then if the user modifies an "Official" wallpaper, a copy is made to the "Custom" one and the user can choose the newly modified, while the original remains intact.

Or, we could just store the official wallpapers on the config directory, and let the user modify them directly.

**Comment by Kangie** [comment]
@Ev1lbl0w You should be right to rebase now. 

Do you want to include this in the 2.2.0 release, or get that out of the way and make a 2.3 release for this change?

**Comment by Ev1lbl0w** [comment]
> Do you want to include this in the 2.2.0 release, or get that out of the way and make a 2.3 release for this change?

Well, since this is a potential breaking change, I'd prefer it to be included on the 2.2 release.

I'll start the rebase now, hopefully it's ready for approval/merge within a few hours.

**Comment by Ev1lbl0w** [comment]
Alright, successfully rebased with the new build system. The only change I had to make was to the `meson.build` files, as the wallpaper creator now depends on `Paths.vala` from the main source code; so I had to alter the build files to be able to include that file.

File: src/OnScreen/BackgroundWindow.vala [hunk 1]
```diff
@@ -387,7 +387,7 @@ namespace Komorebi.OnScreen {
 
 			wallpaperActor.set_content(wallpaperImage);
 
-			wallpaperPixbuf = new Gdk.Pixbuf.from_file_at_scale(@"$package_datadir/$wallpaperName/wallpaper.jpg",
+			wallpaperPixbuf = new Gdk.Pixbuf.from_file_at_scale(@"$wallpaperPath/wallpaper.jpg",
 																scaleWidth, scaleHeight, false);
```
Comment (Kangie) [inline src/OnScreen/BackgroundWindow.vala:391]: I know that you didn't cause this, but I'll get you to fix up the indentation here.

File: src/Paths.vala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		if(wallpaperPaths == null) {
+			wallpaperPaths = {
+				GLib.Path.build_filename(getConfigDir(), "wallpapers"),
+				Config.package_datadir
+			};
+		}
+
+		return wallpaperPaths;
+	}
+}
```
Comment (Kangie) [inline src/Paths.vala:49]: Nitpick: EOF Newline please. :)

File: src/OnScreen/BackgroundWindow.vala (hunk 1 above)
Comment (Ev1lbl0w) [reply src/OnScreen/BackgroundWindow.vala]: Totally right :stuck_out_tongue: fixed

**Comment by Ev1lbl0w** [comment]
*Okidoki*, I fixed the nitpicks, and I'll merge :tada:
